core/state/statedb.go

```diff
@@ -430,9 +430,11 @@ func (s *StateDB) SubBalance(addr common.Address, amount *big.Int) {
 func (s *StateDB) SetBalance(addr common.Address, amount *big.Int) {
 	stateObject := s.GetOrNewStateObject(addr)
 	if stateObject != nil {
-		prevBalance := stateObject.Balance()
-		s.unexpectedBalanceDelta.Add(s.unexpectedBalanceDelta, amount)
-		s.unexpectedBalanceDelta.Sub(s.unexpectedBalanceDelta, prevBalance)
+		if amount != nil {
+			prevBalance := stateObject.Balance()
+			s.unexpectedBalanceDelta.Add(s.unexpectedBalanceDelta, amount)
+			s.unexpectedBalanceDelta.Sub(s.unexpectedBalanceDelta, prevBalance)
```

Comment:
I'm curious how this got hit, but I think we want to treat amount as zero if it's nil for the purposes of the `unexpectedBalanceDelta` computation. That's what `newStateObject` seems to do.

Reply:
Changed it to treat nil amount as zero.